### ISSUE TITLE
SvelteKit generated $types (PageData/ActionData) resolve to {} — false 'does not exist on type {}' errors on data.*/form.*

### ISSUE THREAD

**Issue by razilmsalam** [harshmandan/svelte-check-native#44]
## Summary

On a SvelteKit 2 / Svelte 5 project, `svelte-check-native` reports a flood of false errors because SvelteKit's **generated `./$types`** (`PageData`, `ActionData`, `PageProps`) resolve to `{}`. Every `data.*` / `form.*` access then fails with `Property 'x' does not exist on type '{}'`. The official `svelte-check` reports **0 errors** on the same tree.

## Environment

- `svelte-check-native` 1.2.0
- `typescript` 7.0.2 (also happens with `@typescript/native-preview`)
- SvelteKit 2 (`@sveltejs/kit`), Svelte 5 (runes)
- Node 25.3.0, macOS
- `svelte-kit sync` was run first, so `.svelte-kit/types/**/$types.d.ts` exist.

## Repro

A standard SvelteKit route whose page reads typed load data:

```svelte
<!-- src/routes/foo/+page.svelte -->
<script lang="ts">
  import type { PageProps } from './$types';
  let { data }: PageProps = $props();
</script>

{data.something}
```

with a matching `+page.server.ts` returning `{ something: ... }` from `load`.

Run `npx svelte-check-native`.

## Actual

```
ERROR "src/routes/foo/+page.svelte"  "Property 'something' does not exist on type '{}'."
```

Across a real app this produced **164 errors in 21 files** — all of the shape `Property '<loaded key>' does not exist on type '{}'` (or, for `+page.server.ts`, `... on type 'Omit<{}, never>'`), plus downstream `Parameter 'x' implicitly has an 'any' type` where a callback destructures the now-`{}` data. It also checked **112 files** vs the official checker's **654** on the same project, suggesting part of the type/route graph isn't being picked up.

## Expected

`PageData` / `ActionData` should be inferred from the route's `load` / actions return types via the generated `.svelte-kit/types` proxy modules, exactly as the official `svelte-check` does — i.e. **0 errors** on a project the official checker passes clean.

## Notes

The generated proxy `$types.d.ts` files re-export types derived from the `load` function (roughly `type PageData = Expand<Awaited<ReturnType<typeof load>>>`). It looks like the native checker isn't resolving those generated modules (or the `$types` path alias into `.svelte-kit/types`), so `PageData` falls back to `{}`.

This is the one blocker to using `svelte-check-native` as a drop-in gate for a SvelteKit app on TS 7 — otherwise it runs great and is genuinely fast. Happy to provide a minimal repro repo if useful.

**Comment by razilmsalam** [comment]
## Correction — this is not a bug in svelte-check-native. Closing.

I dug into this properly (built the tool, traced the emit, ran filesystem forensics on my own repro) and the report is wrong. Owning it and closing, with the findings in case they're useful:

**Both suspected failure modes are fine at HEAD:**
- The explicit `let { data }: PageProps = $props()` runes annotation flows through the emit intact and resolves end-to-end. A deliberate `data.doesNotExist` probe fires TS2339 with the *fully inferred* layout-merged type — identical to upstream.
- The generated `$types.d.ts` → `proxy+page.server.js` import chain resolves correctly through the mirror + `rootDirs` union.

On the very project I filed this from (the 654-file app), a cold-cache run with `typescript@7.0.2` + `svelte-check-native@1.2.0` reports **0 errors**.

**What actually happened:** my failing run hit **stale `.svelte-kit/types`** — I had checked out a different branch and my `svelte-kit sync` was piped to /dev/null and didn't take effect, so the generated types didn't match the tree. The "official svelte-check is clean on the same tree" comparison ran through `npm run check`, whose script begins with `svelte-kit sync` — so it silently repaired the staleness before checking. Not the same tree state. Reproducing the stale state makes **upstream `svelte-check` flood the exact same way** — so the tool is at parity, which is its stated bar.

Sorry for the noise. Two genuinely useful things did fall out, filed/offered separately:
1. The explicit `PageProps`/proxy-chain path currently has **no lock-in fixture** — I'll PR a passing regression fixture so it stays covered.
2. The `FILES` denominator diverges from default `svelte-check` on this project (113 vs 654 — svelte/kit files vs all program files). If that's expected vs `--tsgo` mode only, feel free to ignore; otherwise happy to file details.

A possible UX takeaway (maintainer's call, since it diverges from upstream parity): a warning when `src/routes` is newer than `.svelte-kit/types` would have caught my mistake instantly.
